### PR TITLE
AB#84665 Revenue funding view and conditionals workflow

### PR DESCRIPTION
**Pull request by dawid-pawulski**
image

### CODE REVIEW

File: src/HE.Investment.AHP.WWW.Tests/Views/HomeTypes/RevenueFundingTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+namespace HE.Investment.AHP.WWW.Tests.Views.HomeTypes;
+
+public class RevenueFundingTests : HomeTypesTestBase
+{
+    private const string ViewPath = "/Views/HomeTypes/RevenueFunding.cshtml";
+
+    private static readonly RevenueFundingModel Model = new("My application", "My homes");
+
+    [Fact]
+    public async Task ShouldRenderViewWithCheckboxes()
```
Comment (Carq) [inline src/HE.Investment.AHP.WWW.Tests/Views/HomeTypes/RevenueFundingTests.cs:13]: It would be nice to test if specific checkboxes are checked.

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/HomeTypes/Crm/HomeTypeConditionalsMapper.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+using HE.Investment.AHP.Contract.HomeTypes;
+using HE.Investment.AHP.Domain.HomeTypes.Entities;
+
+namespace HE.Investment.AHP.Domain.HomeTypes.Crm;
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/HomeTypes/Crm/HomeTypeConditionalsMapper.cs:4]: this is not CRM mapper but  DTO. Make it private static method in Query Handler or move this class to some other namespace

File: src/HE.Investment.AHP.Contract/HomeTypes/SupportedHousingInformation.cs [hunk 3]
```diff
@@ -7,4 +7,5 @@ public record SupportedHousingInformation(
     string? HomeTypeName,
     YesNoType LocalCommissioningBodiesConsulted,
     YesNoType ShortStayAccommodation,
-    RevenueFundingType RevenueFundingType);
+    RevenueFundingType RevenueFundingType,
+    IList<RevenueFundingSourceType>? RevenueFundingSources);
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Contract/HomeTypes/SupportedHousingInformation.cs:11]: IMO it is better to use non nullable collection and set it to empty collection when nothing is selected, so we'll protect against nulls

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW.Tests/Views/HomeTypes/RevenueFundingTests.cs (hunk 1 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.WWW.Tests/Views/HomeTypes/RevenueFundingTests.cs:12]: done

File: src/HE.Investment.AHP.Domain/HomeTypes/Crm/HomeTypeConditionalsMapper.cs (hunk 2 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.Domain/HomeTypes/Crm/HomeTypeConditionalsMapper.cs]: done

File: src/HE.Investment.AHP.Contract/HomeTypes/SupportedHousingInformation.cs (hunk 3 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.Contract/HomeTypes/SupportedHousingInformation.cs]: done

**Pull request merged**